src/daos/Airtable/CanvasCoursesDao.ts

```diff
@@ -0,0 +1,57 @@
+import Airtable, { FieldSet, Records } from "airtable";
+import { AirtableBase } from "airtable/lib/airtable_base";
+
+class CanvasCoursesDao {
+  private api_key: string;
+  private base_id: string;
+  public airtable: AirtableBase;
+
+  constructor() {
+    if (!process.env.AT_API_KEY) {
+      throw new Error("Missing Airtable API key");
+    }
+    if (!process.env.AT_SMT_BASE_ID) {
+      throw new Error("Missing Airtable base ID");
+    }
+
+    this.api_key = process.env.AT_API_KEY;
+    this.base_id = process.env.AT_SMT_BASE_ID;
+    this.airtable = new Airtable({ apiKey: this.api_key }).base(this.base_id);
+  }
+
+  /**
+   *
+   */
+  public async getAll(): Promise<Records<FieldSet>> {
+    const courses = await this.airtable("Labs - Courses")
+      .select({
+        view: "Grid view",
+      })
+      .all();
+
+    return courses;
+  }
+
+  /**
+   *  @param role
+   */
+  public async getObjectiveCourseByRole(role: string): Promise<number | null> {
+    const courses = await this.airtable("Labs - Courses")
+      .select({
+        view: "Objective Courses",
+        maxRecords: 1,
+        filterByFormula: `{Role} = "${role}"`,
+      })
+      .all();
+
+    if (courses.length) {
+      if (courses[0].fields) {
+        return courses[0].fields["Course ID"] as number;
+      }
+    }
+
+    return null; // TODO: Do we want to return null or [] ?
```

Comment:
Were you planning on getting the whole object? Or just the courseId?

Reply:
Just the ID—[renamed this to "getObjectivesCourseIDByRole" and handed null return](https://github.com/Lambda-School-Labs/labs-be/pull/14/commits/c7176dfd1c173454f2bcace786a27b8a80c998be)